### PR TITLE
removes slot reservations for disconnected players after 5 minutes

### PR DESCRIPTION
**Pull request by PaulRitter**
what it says on the tin

**untested** bc i dont have time :P

### CODE REVIEW

**Comment by moonheart08** [comment]
Any particular reason? Letting them keep the slot doesn't really hurt anything.

File: Content.Server/GameTicking/GameTicker.Player.cs [hunk 1]
```diff
@@ -101,6 +108,11 @@ private void PlayerStatusChanged(object? sender, SessionStatusEventArgs args)
 
                     _chatManager.SendAdminAnnouncement(Loc.GetString("player-leave-message", ("name", args.Session.Name)));
 
+                    if(_decayCancellationTokens.TryGetValue(session.UserId, out var token))
+                        token.Cancel();
+                    token = _decayCancellationTokens[session.UserId] = new CancellationTokenSource();
+                    Timer.Spawn(TimeSpan.FromMinutes(5), () => _playersInGame.Remove(session.UserId), token.Token);
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/GameTicker.Player.cs:114]: cvar pwease

**Comment by metalgearsloth** [comment]
> Any particular reason? Letting them keep the slot doesn't really hurt anything.

If the round goes on an hour and half the pop has left the slots should probably have opened up by that stage? Unless I'm misinterpreting how this works.

**Comment by moonheart08** [comment]
> > Any particular reason? Letting them keep the slot doesn't really hurt anything.
> 
> If the round goes on an hour and half the pop has left the slots should probably have opened up by that stage? Unless I'm misinterpreting how this works.

The system does not reserve a slot. Other players can rejoin in the meantime. It just grants an exception to the player cap.

**Comment by mirrorcult** [comment]
yeah i dont really see why this is necessary

**Comment by PaulRitter** [comment]
> > > Any particular reason? Letting them keep the slot doesn't really hurt anything.
> > 
> > 
> > If the round goes on an hour and half the pop has left the slots should probably have opened up by that stage? Unless I'm misinterpreting how this works.
> 
> The system does not reserve a slot. Other players can rejoin in the meantime. It just grants an exception to the player cap. Looking at the metric, the most the cap has ever been exceeded on USW (our busiest server) is by 3 players. I don't think it's a huge deal to let them keep it til round end and lets in more players than if they just observed the whole time instead.

i misunderstood it then. from what i could tell, it would reserve the slot, not just let people bypass the cap

**Pull request closed without merging**